.github/workflows/elixir.yml

```diff
@@ -20,13 +19,36 @@ jobs:
         otp-version: 22.2 # Define the OTP version [required]
     - name: Check formatting
       run: mix format --check-formatted
+
+    - name: Cache Dependencies
+      id: mix-cache
+      uses: actions/cache@v2
+      with:
+        path: deps
+        key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+        restore-keys: |
+          ${{ runner.os }}-mix-
     - name: Install Dependencies
+      if: steps.mix-cache.outputs.cache-hit != 'true'
       run: mix deps.get
+
+    - name: Cache Build
+      id: build-cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          _build
+          !_build/*/lib/signal_tower
+        key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+        restore-keys: |
+          ${{ runner.os }}-build-
     - name: Compile Dependencies
       run: mix deps.compile
     - name: Compile and check for warnings
       run: mix compile --warnings-as-errors
+
     - name: Run Tests
       run: mix test
+
     - name: Run dialyzer static code analysis
```

Comment:
So caching for dialyzer will come in a subsequent commit? This is the actual thing that takes much time on the first run...

Reply:
No, the files that are created by dialyzer are put into the `_build/dev/` directory, which is cached (except for the actual signal_tower files) already in the Cache Build step. So they will be available in the second run and will be picked up.

Reply:
Ah, but dialyzer is using the Elixir and OTP versions defined above, is it? Then we might habe to add these versions to the cache key as well, to invalidate it, if the Elixir or OTP Version changed.

Reply:
What do you mean with "defined above"? I don't see elixir or otp versions in this file? Maybe you can simply add an empty commit to this pull request so that we can see if dialyzer has become fast? You can later remove it with a force push?

Reply:
By "above" I meant: the Elixir & OTP version is already defined in the "Setup elixir" step in the [current workflow file, line 19-20](https://github.com/palavatv/signaltower/blob/master/.github/workflows/elixir.yml#L19-L20)

But you can ignore my comment about having to add the versions to the cache key, because the PLT files of dialyzer include the versions in their name, i.e. old ones will not be picked up but created newly if the versions  in the setup-elixir step changed.

Re checking whether the changes work: you won’ŧ be able to see that in this PR, because the action that is run for the PR is always the already available in target branch (here: master). 

That’s why I checked this in a test PR in my signaltower fork: https://github.com/paulgoetze/signaltower/actions. This worked, if you check the latest 2 runs, you can see the difference. The older one is the initial one, which created the cache for the deps & build. For the most recent one I just ran a `git --amend --no-edit` and force pushed. It picked up this cache and let dialyzer finish in some seconds.

Sorry, if I was a bit unclear about that :slightly_smiling_face:.

Reply:
And indeed it already used the cache for my force push, nice...wouldn’t have thought that.